src/renderer/components/uielements/input/InputBigNumber.tsx

```diff
@@ -87,12 +69,32 @@ export const InputBigNumber: React.FC<Props> = (props: Props): JSX.Element => {
     onBlurHandler()
   }, [onBlurHandler])
 
-  const onChangeHandler = useCallback(({ target }: React.ChangeEvent<HTMLInputElement>) => {
-    const { value: newValue } = target
-    if (validInputValue(newValue)) {
-      setEnteredValue(O.some(newValue))
-    }
-  }, [])
+  const onChangeHandler = useCallback(
+    ({ target }: React.ChangeEvent<HTMLInputElement>) => {
+      const { value: newValue } = target
+      if (validInputValue(newValue)) {
+        const valueToBroadcast = FP.pipe(
+          O.some(newValue),
+          // ignore empty input
+          O.filter((v) => v !== ''),
+          O.alt(() => '0'),
+          // format value
+          O.map((v) => fixedBN(v, decimal || 5)),
```

Comment:
[nitpick] `decimal` is available by default (see line 19 `const { decimal = 2, ...} = props `)